diff --git a/node_modules/@makeswift/runtime/dist/cjs/next/api-handler/handlers/proxy-preview-mode.js b/node_modules/@makeswift/runtime/dist/cjs/next/api-handler/handlers/proxy-preview-mode.js
index 3f3107f..5a7d725 100644
--- a/node_modules/@makeswift/runtime/dist/cjs/next/api-handler/handlers/proxy-preview-mode.js
+++ b/node_modules/@makeswift/runtime/dist/cjs/next/api-handler/handlers/proxy-preview-mode.js
@@ -26,7 +26,7 @@ var import_http_proxy = require("http-proxy");
 var import_set_cookie_parser = require("set-cookie-parser");
 var import_preview_mode = require("../../preview-mode");
 async function proxyPreviewMode(req, res, { apiKey }) {
-  const previewModeProxy = (0, import_http_proxy.createProxyServer)();
+  const previewModeProxy = (0, import_http_proxy.createProxyServer)({ xfwd: true });
   const NextRequestMetaSymbol = Reflect.ownKeys(req).find(
     (key) => key.toString() === "Symbol(NextRequestMeta)" || key.toString() === "Symbol(NextInternalRequestMeta)"
   );
diff --git a/node_modules/@makeswift/runtime/dist/esm/next/api-handler/handlers/proxy-preview-mode.js b/node_modules/@makeswift/runtime/dist/esm/next/api-handler/handlers/proxy-preview-mode.js
index 346181c..3b10208 100644
--- a/node_modules/@makeswift/runtime/dist/esm/next/api-handler/handlers/proxy-preview-mode.js
+++ b/node_modules/@makeswift/runtime/dist/esm/next/api-handler/handlers/proxy-preview-mode.js
@@ -3,7 +3,7 @@ import { createProxyServer } from "http-proxy";
 import { parse } from "set-cookie-parser";
 import { MakeswiftSiteVersion } from "../../preview-mode";
 async function proxyPreviewMode(req, res, { apiKey }) {
-  const previewModeProxy = createProxyServer();
+  const previewModeProxy = createProxyServer({ xfwd: true });
   const NextRequestMetaSymbol = Reflect.ownKeys(req).find(
     (key) => key.toString() === "Symbol(NextRequestMeta)" || key.toString() === "Symbol(NextInternalRequestMeta)"
   );
